modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx

```diff
@@ -0,0 +1,27 @@
+import { Box, Button, Text } from "@mantine/core";
+import { FaPlus } from "react-icons/fa";
+
+import { useAppSelector } from "@/shared/redux/hooks";
+import { authenticatedUserSelector } from "@/shared/redux/reducers/user.reducer";
+import { ERole } from "@/shared/typedefs";
+
+import classes from "./AllClassrooms.module.css";
+
+const AllClassrooms = () => {
+  const user = useAppSelector(authenticatedUserSelector);
+  return (
+    <Box mih={"91vh"} pos={"relative"}>
+      {user.claim === ERole.TEACHER ? (
```

Comment:
```suggestion
      {user?.claim === ERole.TEACHER ? (
```

[NIT]: Best to use optional chaining. If the user logs out from this page, the user object would be empty and attempting to access user.claim would throw an error.